components/sup/src/ctl_gateway/server.rs

```diff
@@ -461,46 +468,42 @@ enum SrvHandlerState {
 ///
 /// New connections will be authenticated using `secret_key`. Messages from the main thread
-/// will be sent over the channel `mgr_tx`.
-pub fn run(listen_addr: SocketAddr, secret_key: String, mgr_tx: MgrSender) {
-    thread::Builder::new().name("ctl-gateway".to_string())
-                          .spawn(move || {
-                              let mut core = reactor::Core::new().unwrap();
-                              let handle = core.handle();
-                              let listener = TcpListener::bind(&listen_addr).unwrap();
-                              let state = SrvState { secret_key, mgr_tx };
-                              let state = Rc::new(RefCell::new(state));
-                              let clients =
-                                  listener.incoming().map(|socket| {
-                                                         let addr = socket.peer_addr().unwrap();
-                                                         let io =
-                                                             Framed::new(socket, SrvCodec::new());
-                                                         (
-                    Client {
-                        handle: handle.clone(),
-                        state: state.clone(),
-                    }
-                    .serve(io),
-                    addr,
-                )
-                                                     });
-                              let server = clients.for_each(|(client, addr)| {
-                                                      handle.spawn(client.then(move |res| {
-                                                                       debug!("DISCONNECTED from \
-                                                                               {:?} with result \
-                                                                               {:?}",
-                                                                              addr, res);
-                                                                       future::ok(())
-                                                                   }));
-                                                      Ok(())
-                                                  });
-                              core.run(server)
-                          })
-                          .expect("ctl-gateway thread start failure");
+/// will be sent over the channel `mgr_sender`.
+pub fn run(listen_addr: SocketAddr, secret_key: String, mgr_sender: MgrSender) {
+    let tb = thread::Builder::new().name("ctl-gateway".to_string());
+    tb.spawn(move || {
+          let mut core = reactor::Core::new().unwrap();
+          let handle = core.handle();
+          let state = SrvState { secret_key,
+                                 mgr_sender };
+          let state = Rc::new(RefCell::new(state));
+          let server =
+              TcpListener::bind(&listen_addr).unwrap()
```

Comment:
I realize this was existing code but is it worth at least an `expect` here?  I know we've gone to great lengths to handle binding errors gracefully in other parts of the code.

Reply:
Probably can't hurt! 👍